### DISCUSSION TITLE
pGlyph > fg_layers(asDict=True) crashes FontLab

### DISCUSSION THREAD

**Discussion by alex-phil** [kateliev/TypeRig#62] [General]
Vassil, please take a look at this script:
```
from typerig.proxy.fl.objects.glyph import pGlyph

layers_dict = pGlyph().fg_layers(asDict=True)
print(layers_dict)
```
When you run the script on a glyph that has at least one non-empty layer, FontLab 8 (v8244) crashes immediately. When you run the script on a glyph with empty layers only, FontLab 8 does not crash, and the output is correct. Thank you for looking into this.

**Comment by kateliev** [comment]
@alex-phil you are right, but unfortunately this is totally not working. It all looks like a FL8 bug where even indexing the returned list of fg_layers causes the app to crash! The operation fgGlyph.layers[index] is not possible thus creating a dict of it is also very much impossible...

 But please indulge my curiosity  ... why whoud you need this .fg_layers() call in first place.. when you have .layers() that will give you back the list of necessary layers (flLayer) to work with. To be honest i do not even remember why i still keep that .fg_layers() because there is absolutely nothing one needs to work with in the FG domain. As a rule of the thumb, you cannot work your way trough it. If you need something done to the FG part of FL, you need to access the FL part of it... convert it to FG... work with it and then covert it back. Otherwise you will crash the app. On the FG layer.. the only thing that is modifiable is Kerning.. and you have to be very careful with it :) So please let me know if you need something special in the FG domain and i might offer you a workaround....

**Comment by kateliev** [comment]
@alex-phil this is now a discussion. Feel free to post below :)

**Comment by alex-phil** [comment]
Vassil, please don’t mind. It’s just that I recently started working more systematically through your proxy system, taking notes as I go, and I noticed that I got consistent crashes at this point. Which made me wonder why that might be the case. So thank you for the explanations above. They will help me distinguish FL8 bugs from possible amendments to your proxy system.

**Comment by kateliev** [comment]
@alex-phil i do not believe it is a bug per se... more that the FG system was not intended to be used that way... to me trying to use FG is more like trying to jump start a modern electronically controlled car by hand cranking the engine :) In theory it should work - it is an internal combustion engine after all - but in practice it never will as the electronics will never allow it :)
